flight/Modules/Battery/battery.c

```diff
@@ -157,12 +167,19 @@ static void batteryTask(void * parameters)
 
 			// A negative result indicates an error (PIOS_ADC_GetChannelVolt returns -1 on error)
 			if(flightBatteryData.Current < 0.0f)
-				flightBatteryData.Current = 0.0f;
+				adc_pin_invalid = true;
```

Comment:
Likewise, would be cleaner to check the ```<``` just once, and not to compute the actual current if it's not necessary.